### PR TITLE
Handle proxy_uri consistently in iotedge check

### PR DESCRIPTION
**Pull request by scb01**
Adjust iotedge check to look for proxy-uri information first on the
command line, next from the aziot-edged settings under [agent.env], and
finally from the enviroment.

With this, iotedge check will consistently pass proxy_uri info to the
`AziotEdgedVersion` and `ContainerConnectUpStream` checks.

### CODE REVIEW

File: edgelet/iotedge/src/check/checks/container_connect_upstream.rs [hunk 1]
```diff
@@ -166,14 +166,8 @@ impl ContainerConnectUpstream {
         }
 
         if &port == "443" {
-            let proxy = settings
-                .agent()
-                .env()
-                .get("https_proxy")
-                .map(std::string::String::as_str);
-            self.proxy = proxy.map(ToOwned::to_owned);
```
Comment (damonbarry) [inline edgelet/iotedge/src/check/checks/container_connect_upstream.rs:174]: In your changes, `self.proxy` is no longer being assigned. However, it looks as though `self.proxy`, though assigned, wasn't being used for anything anyway. Maybe it should be removed from `struct ContainerConnectUpstream`?

File: edgelet/iotedge/src/check/mod.rs [hunk 2]
```diff
@@ -890,4 +902,64 @@ mod tests {
             ),
         }
     }
+
+    #[test]
+    fn pickup_proxy_uri_from_the_right_place() {
+        //Setup the environment
+        let env_proxy_uri = "https://environment:123";
+        std::env::set_var("HTTPS_PROXY", env_proxy_uri);
```
Comment (damonbarry) [inline edgelet/iotedge/src/check/mod.rs:910]: Test "https_proxy" (lowercase) env var too?

File: edgelet/edgelet-docker/test/linux/sample_settings_with_proxy_uri.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+name = "edgeAgent"
+type = "docker"
+
+[agent.config]
+image = "microsoft/azureiotedge-agent:1.0"
+
+[agent.env]
+abc = "value1"
+acd = "value2"
+"https_proxy" = "https://config:123"
```
[the remark this answers is not in the document]
Comment (arsing) [reply edgelet/edgelet-docker/test/linux/sample_settings_with_proxy_uri.toml:14]: Yes, both are valid TOML, but there's no reason it should be in quotes so remove it for consistency.

File: edgelet/iotedge/src/check/checks/container_connect_upstream.rs (hunk 1 above)
Comment (arsing) [reply edgelet/iotedge/src/check/checks/container_connect_upstream.rs:174]: >wasn't being used for anything anyway.

It was there to be serialized into the JSON output.

File: edgelet/iotedge/src/check/mod.rs [hunk 4]
```diff
@@ -666,6 +660,24 @@ impl Check {
     }
 }
 
+fn get_proxy_uri(arg: Option<String>) -> Option<String> {
+    //If proxy address was passed in as command line argument, we are good
```
Comment (arsing) [inline edgelet/iotedge/src/check/mod.rs:664]: Missing space after the `//` in all these lines.

File: edgelet/iotedge/src/check/mod.rs (hunk 2 above)
Comment (arsing) [reply edgelet/iotedge/src/check/mod.rs:910]: Yes, needs to check both, like the old code used to.

File: edgelet/iotedge/src/check/checks/container_connect_upstream.rs (hunk 1 above)
Comment (damonbarry) [reply edgelet/iotedge/src/check/checks/container_connect_upstream.rs:174]: Ah, I see. So @scb01 you need to bring back that assignment.

Comment (scb01) [reply edgelet/iotedge/src/check/checks/container_connect_upstream.rs:174]: Makes sense. Addressed.

Comment (damonbarry) [reply edgelet/iotedge/src/check/checks/container_connect_upstream.rs:174]: @scb01 Did you confirm that the JSON output from the check now contains the proxy address?

Comment (scb01) [reply edgelet/iotedge/src/check/checks/container_connect_upstream.rs:174]: Yep. Ran `iotedge check --output json` locally and verified that the output contains the configured value for proxy-uri ( or null if proxy-uri is not configured)

**Review by damonbarry — DISMISSED** [review]

**Review by damonbarry — DISMISSED** [review]

**Review by damonbarry — APPROVED** [review]

**Pull request merged**